File: bestia-zoneserver/src/main/java/net/bestia/zoneserver/map/path/PointEstimator.java
package net.bestia.zoneserver.map.path;

import net.bestia.model.geometry.Point;

/**
 * Heuristically estimates the distances of {@link Point}s.
 * 
 * @author Thomas Felix
 *
 */
public class PointEstimator implements HeuristicEstimator<Point> {

	@Override
	public float getDistance(Point current, Point target) {
		return (float) current.getDistance(target);
	}

}

File: bestia-zoneserver/src/main/kotlin/net/bestia/zoneserver/map/path/PointEstimator.kt
package net.bestia.zoneserver.map.path

import net.bestia.model.geometry.Vec3

/**
 * Heuristically estimates the distances of [Vec3]s.
 *
 * @author Thomas Felix
 */
class PointEstimator : HeuristicEstimator<Vec3> {

  override fun getDistance(current: Vec3, target: Vec3): Float {
    return current.getDistance(target).toFloat()
  }
}
